benchmark_data_loading/dataset_helpers.py

```diff
@@ -102,28 +103,51 @@ def _make_dp_from_tars(*, root, archive_size):
     return dp
 
 
-def _make_webdataset(*, root, archive_size):
-    archives = Path(root).glob(f"archive_{archive_size}*.tar")
-    archives = [str(a) for a in archives]
-    return wds.WebDataset(archives)  # This will read and load the data as bytes
-
-
 def make_dp(*, root, archive=None, archive_content=None, archive_size=500):
     if archive in ("pickle", "torch"):
         dp = _make_dp_from_archive(
             root=root, archive=archive, archive_content=archive_content, archive_size=archive_size
         )
     elif archive == "tar":
         dp = _make_dp_from_tars(root=root, archive_size=archive_size)
-    elif archive == "webdataset":
-        dp = _make_webdataset(root=root, archive_size=archive_size)
     else:
         dp = _make_dp_from_image_folder(root=root)
 
     dp = LenSetter(dp)
     return dp
 
 
+def make_webdataset_dataloader(*, root, encoded=True, archive_size=500):
+    def preprocess(sample):
+        encoded_image = bytesio_to_tensor(io.BytesIO(sample["jpeg"]))
+        label = int(str(sample["__key__"].split(os.sep)[0]))
+        return encoded_image, label
+
+    archives = Path(root).glob(f"archive_{archive_size}*.tar")
+    archives = [str(a) for a in archives]
+
+    dataset = wds.WebDataset(archives).map(preprocess)
+
+    if not encoded:
+        dataset = dataset.map_tuple(decode, None)
+
+    if args.num_workers == 0:
+        return dataset.shuffle(INFINITE_BUFFER_SIZE, initial=INFINITE_BUFFER_SIZE)
+
+    return (
+        wds.WebLoader(
+            dataset,
+            batch_size=None,
+            num_workers=args.num_workers,
+        )
+        .shuffle(INFINITE_BUFFER_SIZE, initial=INFINITE_BUFFER_SIZE)
```

Comment:
With this `shuffle` here, running `bench_data_reading.py` I get weird errors like

```
RuntimeError: unable to mmap 2072 bytes from file <filename not specified>: Cannot allocate memory (12)
```

Could you see if the same happens for you? Note that you need `num_workers > 0` to use to get o this branch.

Reply:
Problem magically went away. Cannot reproduce anymore.